### DISCUSSION TITLE
Set headers to None to exclude specific default headers

### DISCUSSION THREAD

**Discussion by GeekDuanLian** [encode/httpx#1662] [Ideas]
Just like requests

`s = Session(headers={'authorization': 'xxx'})`

But you want to visit an external website, the authorization cannot be leaked

`s.get('xxx', headers={'authorization': None})`

This way to exclude the default headers

**Comment by lovelydinosaur** [comment]
I'm not convinced if we'd want to support this usage or not.
Having a special casing behaviour for `None` in this way seems a bit brittle.

(?)

**Reply by GeekDuanLian** [reply]
Are there other ways to exclude default headers?
At least it should have this function, and requests do this.
